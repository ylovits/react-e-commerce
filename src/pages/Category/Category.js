import React, { Fragment, useState, useEffect } from "react";
import { withRouter } from "react-router-dom";
import Spinner from "../../components/Spinner/Spinner";
import ProductTeaser from "../../components/ProductTeaser/ProductTeaser.js";
import "./Category.scss";

const Category = ({ match, history }) => {
  const [products, setProducts] = useState([]);
  const [loading, setLoading] = useState(true);
  const productsjson = require("../../fakeapi/products.json");

  useEffect(() => {
    setLoading(true);
    var categoryProds = productsjson.filter(product =>
      product.category
        .toLowerCase()
        .includes(match.params.categoryId.toLowerCase())
    );
    setProducts(categoryProds);
    // should be await async when the api is real http call, not setTimeout
    setTimeout(() => setLoading(false), 1500);
    // eslint-disable-next-line
  }, []);

  const categExists = categName => {
    let i = null;
    for (i = 0; productsjson.length > i; i += 1) {
      if (productsjson[i].category.toLowerCase() === categName) {
        return true;
      }
    }
    return false;
  };

  if (categExists(match.params.categoryId.toLowerCase())) {
    if (loading) {
      return <Spinner />;
    }
    return (
      <Fragment>
        <div className="row category">
          {products.map(product => {
            return <ProductTeaser key={product.MPN} product={product} />;
          })}
        </div>
      </Fragment>
    );
  } else {
    history.push("/404");
  }
};

export default withRouter(Category);
