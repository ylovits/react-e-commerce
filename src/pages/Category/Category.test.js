import React from "react";
import { shallow } from "enzyme";
import Category from "./Category";

it("<Category /> rendered", () => {
  expect(shallow(<Category />)).toMatchSnapshot();
});
