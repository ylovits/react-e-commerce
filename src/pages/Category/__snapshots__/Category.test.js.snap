// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> rendered 1`] = `ShallowWrapper {}`;
